ext/resourcemanager/optimus.go

```diff
@@ -33,16 +35,43 @@ func NewOptimusResourceManager(resourceManagerConfig config.ResourceManager) (*O
 	if err := mapstructure.Decode(resourceManagerConfig.Config, &conf); err != nil {
 		return nil, fmt.Errorf("error decoding resource manger config: %w", err)
 	}
+
 	if conf.Host == "" {
 		return nil, errors.New("optimus resource manager host is empty")
 	}
+
+	httpClient, err := newHTTPClient(conf.Host)
```

Comment:
Can we also add the code to add http headers from config.

Reply:
I checked that it wasn't possible here. But, we already did put the header under file `ext/resourcemanager/optimus.go` when constructing the request:

```go
for key, value := range o.config.Headers {
  // ...
}
```